llm4pddl/approaches/llm_open_loop_approach.py

```diff
@@ -182,3 +200,61 @@ def _llm_response_to_plan(response: LLMResponse, task: Task) -> Plan:
             action = f"({op} {objects_str})"
             plan.append(action)
         return plan
+
+    def embed_task(self, task: Task,
+                   embedding_model: SentenceTransformer) -> Embedding:
+        """Embeds a task using embedding_model.
+
+        Returns a numpy array.
+        """
+        # note: task_string includes the Q: and A: parts, not just task string
+        task_string = self._create_prompt(task)
+        # this line is a work around for now:
+        task_string = task_string.split('\n')[1:-2][0]
+        embedding = embedding_model.encode(task_string)
+        return embedding
+
+    def embed_tasks(self, tasks: List[Task]) -> List[Embedding]:
+        """"Embeds a list of tasks.
+
+        Returns a list of embeddings with indices corresponding to its
+        task.
+        """
+        embedding_model = SentenceTransformer(FLAGS.embedding_model_name)
+        embeddings = [self.embed_task(task, embedding_model) for task in tasks]
+        return embeddings
+
+    def make_embeddings_mapping(self, embeddings: List[Embedding],
+                                dataset: Dataset) -> List[Dict[str, Any]]:
+        """Makes embeddings mapping for training data."""
+        assert len(embeddings) == len(dataset)
+        return [{
+            'embedding': emb,
+            'datum': datum
+        } for emb, datum in zip(embeddings, dataset)]
+
+    def get_closest_datums(self, task: Task,
+                           embeddings_mapping: List[Dict[str, Any]],
+                           num_closest: int) -> List[Datum]:
+        """Returns the num_train most similar training tasks to the task, in
```

Comment:
num_closest* not num_train, right? I guess these are the same thing right now, but won't necessarily be in the near future